core/domain/rights_manager_test.py

```diff
@@ -849,73 +635,51 @@ def test_setting_rights_of_collection(self):
 
     def test_publishing_and_unpublishing_collection(self):
         self.save_new_default_collection(self.COLLECTION_ID, self.user_id_a)
+        collection_rights = rights_manager.get_collection_rights(
+            self.COLLECTION_ID)
 
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_b).can_play(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_b).can_view(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
+        self.assertFalse(rights_manager.check_can_access_activity(
+            self.user_id_b, self.actions_b, collection_rights))
 
         rights_manager.publish_collection(self.user_id_a, self.COLLECTION_ID)
+        collection_rights = rights_manager.get_collection_rights(
+            self.COLLECTION_ID)
 
-        self.assertTrue(
-            rights_manager.Actor(self.user_id_b).can_play(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertTrue(
-            rights_manager.Actor(self.user_id_b).can_view(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_a).can_unpublish(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
+        self.assertTrue(rights_manager.check_can_access_activity(
+            self.user_id_b, self.actions_b, collection_rights))
```

Comment:
the test is to check if collection is published by rights_manager.publish_collection(...)
If a user cannot view the collection before and can view the collection after this function, then the collection is published.

But added the test back anyway.